src/pages/index.js

```diff
@@ -92,20 +114,73 @@ import { PopupWithImage } from "../components/PopupWithImage";
     `#${addForm.parentElement.id}`,
     handleProfileAddSubmit
   );
-  const userInfo = new UserInfo(profileName, profileDescription);
 
   imagePopup.setEventListeners();
 
+  //preset profile info on load
+  api.getProfileInfo().then((results) => {
+    console.log(results);
+    profileName.textContent = results.name;
+    profileDescription.textContent = results.about;
+    avatarPhoto.src = results.avatar;
+  });
+
   //functions
+  const deletePopup = new confirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        console.log(card._id);
+
+        // api.deleteCard(card._id, card.deleteCard()); // 
+        deletePopup.open();
+        deletePopup.setSubmitAction(() =>{
+          api.deleteCard(card._id).then(() =>{
+            card.deleteCard();
+            deletePopup.close();
+          })
+        })
+
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+
+    galleryList.prepend(modelCard.addCard());
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    api.addCardInfo(name, description, (name, link, id) => {
+      console.log(template);
+      addCard(name, link, template, id);
+    }).then(
+      addPopup.saveButton.textContent = "Saving..."
+    )
   }
+  // editPopup.saveButton.textContent = "Submit";
+  // addPopup.saveButton.textContent = "Submit";
+
+  function handleProfileEditSubmit(inputElements) {
+    // const saveButton = editPopup.querySelector(".modal__save-button");
+    console.log(editPopup)
+    api
+      .setUserInfo(inputElements.name, inputElements.description)
+      .then((result) => {
+        user.setUserInfo({ name: result.name, about: result.about });
+      })
+      .then(
+        editPopup.saveButton.textContent = "Saving..."
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```